### PR TITLE
Replace error string with ErrorList in scheduler validation testing

### PR DESCRIPTION
**Pull request by yuanchen8911**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Replace error string with `field.ErrorList` for more comprehensive testing coverage in scheduler `validation_test.go`. 

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

A follow-up of this PR: https://github.com/kubernetes/kubernetes/pull/112521/files#r984355471 suggested by @alculquicondor. 

Follow-up in a separate PR:
- Fix `LeaderElection` error type: `Invalid` to `Required`

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yuanchen8911** [comment]
I don't think there's any API change in this PR.

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 1]
```diff
@@ -198,15 +199,15 @@ func TestValidateKubeSchedulerConfigurationV1beta2(t *testing.T) {
 
 	extenderDuplicateManagedResource := validConfig.DeepCopy()
 	extenderDuplicateManagedResource.Extenders[0].ManagedResources = []config.ExtenderManagedResource{
-		{Name: "foo", IgnoredByScheduler: false},
-		{Name: "foo", IgnoredByScheduler: false},
+		{Name: "example.com/foo", IgnoredByScheduler: false},
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:202]: why did this have to change?
Did it have an extra validation error?

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 2]
```diff
@@ -225,102 +226,202 @@ func TestValidateKubeSchedulerConfigurationV1beta2(t *testing.T) {
...
+				},
+			},
 		},
 		"bad-resource-name-not-set": {
 			expectedToFail: true,
 			config:         resourceNameNotSet,
-			errorString:    "resourceName is required",
+			wantErrs: field.ErrorList{
+				&field.Error{
+					Type:  field.ErrorTypeInvalid,
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:241]: uhm... we should change this to ErrorTypeRequired, but probably a separate PR, as it's not scheduler code

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 3]
```diff
@@ -225,102 +226,202 @@ func TestValidateKubeSchedulerConfigurationV1beta2(t *testing.T) {
...
+				},
+			},
 		},
 		"bad-resource-namespace-not-set": {
 			expectedToFail: true,
 			config:         resourceNamespaceNotSet,
-			errorString:    "resourceNamespace is required",
+			wantErrs: field.ErrorList{
+				&field.Error{
+					Type:  field.ErrorTypeInvalid,
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:251]: same here

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 4]
```diff
@@ -216,7 +217,7 @@ func TestValidateKubeSchedulerConfigurationV1beta2(t *testing.T) {
 	scenarios := map[string]struct {
 		expectedToFail bool
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:218]: can we get rid of this boolean?
`cmp.Diff` already checks if the errors matches the empty list.

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 5]
```diff
@@ -513,15 +613,15 @@ func TestValidateKubeSchedulerConfigurationV1beta3(t *testing.T) {
...
+		{Name: "example.com/foo", IgnoredByScheduler: false},
+		{Name: "example.com/foo", IgnoredByScheduler: false},
 	}
 
 	extenderDuplicateBind := validConfig.DeepCopy()
 	extenderDuplicateBind.Extenders[0].BindVerb = "foo"
 	extenderDuplicateBind.Extenders = append(extenderDuplicateBind.Extenders, config.Extender{
 		PrioritizeVerb: "prioritize",
-		BindVerb:       "bar",
+		BindVerb:       "foo",
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:624]: why the change here?

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 6]
```diff
@@ -531,7 +631,7 @@ func TestValidateKubeSchedulerConfigurationV1beta3(t *testing.T) {
 	scenarios := map[string]struct {
 		expectedToFail bool
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:632]: same question about the boolean

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 7]
```diff
@@ -850,7 +1050,7 @@ func TestValidateKubeSchedulerConfigurationV1(t *testing.T) {
...
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:1051]: same

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 1 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:202]: foo or bar is an invalid ExtenderResource name. The original version returned additional invalid resource errors.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 4 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:218]: I think so. I'll remove it.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 5 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:624]: Is it supposed to check duplicate name error?

Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:624]: No, this one is to test duplicate `BindVerb`  indicated by the name `extenderDuplicateBind`.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 4 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go]: Deleted all `expectedToFail` bool and check ``errs == nil && diff !="" `  for unexpected success instead.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 7 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go]: deleted it

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 8]
```diff
@@ -198,148 +199,222 @@ func TestValidateKubeSchedulerConfigurationV1beta2(t *testing.T) {
...
 	}
 
 	for name, scenario := range scenarios {
 		t.Run(name, func(t *testing.T) {
 			errs := ValidateKubeSchedulerConfiguration(scenario.config)
-			if errs == nil && scenario.expectedToFail {
+			diff := cmp.Diff(scenario.wantErrs.ToAggregate(), errs, ignoreBadValueDetail)
+			if errs == nil && diff != "" {
 				t.Error("Unexpected success")
 			}
```
[the remark this answers is not in the document]
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:415]: You are right. Just need to check `diff != nil`.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 5 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:599]: You are right. Reverted the change.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 3 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go:246]: Will fix it in a separate PR.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 6 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go]: Deleted it.

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 5 above)
Comment (alculquicondor) [reply pkg/scheduler/apis/config/validation/validation_test.go:596]: did you forget to revert here?

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 9]
```diff
@@ -829,15 +972,15 @@ func TestValidateKubeSchedulerConfigurationV1(t *testing.T) {
...
+		{Name: "example.com/foo", IgnoredByScheduler: false},
+		{Name: "example.com/foo", IgnoredByScheduler: false},
 	}
 
 	extenderDuplicateBind := validConfig.DeepCopy()
 	extenderDuplicateBind.Extenders[0].BindVerb = "foo"
 	extenderDuplicateBind.Extenders = append(extenderDuplicateBind.Extenders, config.Extender{
 		PrioritizeVerb: "prioritize",
-		BindVerb:       "bar",
+		BindVerb:       "foo",
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_test.go:983]: and revert here too

**Comment by alculquicondor** [comment]
Almost there
/approve

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 9 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go]: done

File: pkg/scheduler/apis/config/validation/validation_test.go (hunk 5 above)
Comment (yuanchen8911) [reply pkg/scheduler/apis/config/validation/validation_test.go]: Yes, I forgot to change two versions. Reverted for all.

**Comment by alculquicondor** [comment]
Thanks!
/lgtm
/retest
